docs/building-a-form/common-patterns-for-building-forms.md

```diff
@@ -0,0 +1,298 @@
+    chapter33: { ... },
+    transferOfEntitlement: { ... },
+    veteranFullName: { ... },
+    veteranSocialSecurityNumber: { ... }
+  },
+  other: { ... }
+}
+
+// uiSchema
+{
+  disability: { ... },
+  dic: { ... },
+  chapter31: { ... },
+  ownServiceBenefits: { ... },
+  'view:sponsorServiceOptions': {
+    hideIf: (formData) => /* Some condition here */,
+    chapter35: { ... },
+    chapter33: { ... },
+    transferOfEntitlement: { ... },
+    veteranFullName: { ... },
+    veteranSocialSecurityNumber: { ... }
+  },
+  other: { ... }
+}
+```
+
+From this, the fields in the `view:sponsorServiceOptions` object are moved up one level and sent alongside `dic` and `chapter31`. The back end doesn't see objects with names that start with `view:`, but it gets all fields inside those objects.
+
+## Styling expanded or collapsed fields
+
+To indent or otherwise style fields that are expanded or collapsed with the `expandUnder` option, set a class on the controlling field. For example:
+
+```js
+// uiSchema
+{
+  field1: {
+    'ui:title': 'This field expands/collapses other items',
+    'ui:options': {
+      expandUnderClassNames: 'schemaform-expandUnder-indent'
+    }
+  },
+  field2: {
+    'ui:title': 'This field is controlled by field1'
+    'ui:options': {
+      expandUnder: 'field1'
+    }
+  },
+  field3: {
+    'ui:title': 'This field is controlled by field1'
+    'ui:options': {
+      expandUnder: 'field1'
+    }
+  }
+}
+```
+
+In this example, `schemaform-expandUnder-indent` is applied to the `div` that surrounds `field2` and `field3`, which indents the fields. For additional styling, create a new class and add your own styles.
+
+### Conditionally including a page
+
+The `depends` property determines whether a page is active or not. `depends` can work in a few ways:
+
+```js
+// With an object
+depends: {
+  passPhrase: 'open sesame'
+}
+
+// With an array
+// This will activate the page if any of the items in the array are true. Think || not &&.
+depends: [
+  { passPhrase: 'open sesame' },
+  { passPhrase: 'open up!' }
+]
+
+// With a function
+depends: (formData) => {
+  // return bool, true if page is active, false if page should be skipped
+  return formData.passPhrase === 'open sesame' && formData.codeWord === 'chicken';
+}
+```
+
+For example, given this chapter configuration:
+
+```js
+chapterName: {
+  title: 'Chapter Title',
+  pages: {
+    pageName: {
+      ...
+      schema: {
+        type: 'object',
+        properties: {
+          passPhrase: { type: 'string' }
+        }
+      }
+    }
+    otherPageName: {
+      title: 'Page title',
+      path: 'path/to/page',
+      initialData: {},
+      depends: {
+        passPhrase: 'open sesame'
+      },
+      uiSchema: {},
+      schema: {}
+    }
+  }
+}
+```
+
+If a user types 'open sesame' for the `passPhrase` on the first page, `otherPageName` is active. For any other value, including no value, `otherPageName` is inactive and the page is skipped.
+
```

Comment:
May want to note here that the text of these pass phrases is in clear text in the source (even when it's in a function) so it should not be used as any type of secure login or other authentication mechanism.